File: leetcode/JulyStreak/Binary_Tree_Level_Order_Traversal.java
public class TreeNode {
int val;
TreeNode left;
TreeNode right;
TreeNode() {}
TreeNode(int val) { this.val = val; }
TreeNode(int val, TreeNode left, TreeNode right) {
this.val = val;
this.left = left;
this.right = right;
}
}
 
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        Queue<TreeNode> q = new LinkedList<TreeNode>();
        List<List<Integer>> list = new LinkedList<List<Integer>>();
        if(root == null) return list;
        q.offer(root);
        while(q.size() > 0){
            int count = q.size();
            List<Integer> sublist = new LinkedList<Integer>();
            for(int i =0;i<count;i++){
                if(q.peek().left != null) q.offer(q.peek().left);
                if(q.peek().right != null) q.offer(q.peek().right);
                sublist.add(q.poll().val);
            }
            list.add(sublist);
            
        }
        return list;
    }
}

File: Hacktoberfest2022/Binary Tree Level Order Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) 
    {
        vector<vector<int>>v;
        if(!root) return v;
        queue<TreeNode * > q;
        q.push(root);
        while(!q.empty())
        {
            int s = q.size();
            vector<int>h;
            for(int i = 0; i<s ; i++)
            {
                TreeNode * k = q.front();
                h.push_back(k->val);
                if(k->left) q.push(k->left);
                if(k->right) q.push(k->right);
                q.pop();
            }
            v.push_back(h);
        }
        return v;
    }
};